test/Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -2471,46 +2471,48 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
                 }
             };
 
-            var aborted = new ManualResetEventSlim();
-
             using (var server = new TestServer(async context =>
             {
-                context.RequestAborted.Register(() =>
-                {
-                    aborted.Set();
-                });
+                context.RequestAborted.Register(() => requestAborted.Set());
 
-                context.Response.ContentLength = chunks * chunkSize;
+                context.Response.ContentLength = responseSize;
 
                 for (var i = 0; i < chunks; i++)
                 {
                     await context.Response.WriteAsync(new string('a', chunkSize), context.RequestAborted);
                 }
             }, testContext))
             {
-                using (var connection = server.CreateConnection())
+                using (var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
                 {
-                    await connection.Send(
-                        "GET / HTTP/1.1",
-                        "Host:",
-                        "",
-                        "");
+                    socket.ReceiveBufferSize = 1;
 
-                    Assert.True(aborted.Wait(TimeSpan.FromSeconds(60)));
+                    socket.Connect(new IPEndPoint(IPAddress.Loopback, server.Port));
+                    socket.Send(Encoding.ASCII.GetBytes("GET / HTTP/1.1\r\nHost: \r\n\r\n"));
 
-                    await connection.Receive(
-                        "HTTP/1.1 200 OK",
-                        "");
-                    await connection.ReceiveStartsWith("Date: ");
-                    await connection.Receive(
-                        $"Content-Length: {chunks * chunkSize}",
-                        "",
-                        "");
+                    Assert.True(messageLogged.Wait(TimeSpan.FromSeconds(60)));
```

Comment:
Consider adding a messages when using Assert.True/False so when the assertion fails, we know which one. eg
```c#
Assert.True(messageLogged.Wait(TimeSpan.FromSeconds(60)), "The message should have been received but timed out");
```